api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -165,15 +165,21 @@ message Bootstrap {
   // Configuration for internal processing of stats.
   metrics.v3.StatsConfig stats_config = 13;
 
-  // Optional duration between flushes to configured stats sinks. For
-  // performance reasons Envoy latches counters and only flushes counters and
-  // gauges at a periodic interval. If not specified the default is 5000ms (5
-  // seconds).
-  // Duration must be at least 1ms and at most 5 min.
-  google.protobuf.Duration stats_flush_interval = 7 [(validate.rules).duration = {
-    lt {seconds: 300}
-    gte {nanos: 1000000}
-  }];
+  oneof stats_flush {
```

Comment:
As @jmarantz mentioned v2 is frozen and should not be changed.
In v3 changing the `stats_flush_interval` to oneof will break [backwards compatibility](https://github.com/envoyproxy/envoy/blob/master/api/API_VERSIONING.md#backwards-compatibility).
You should be able to annotate this as oneof_promotion (see example [here](https://github.com/envoyproxy/envoy/blob/de222c3acf1fbf124ca44f50495fc4b8e9aba525/tools/testdata/protoxform/envoy/v2/oneof.proto) and [here](https://github.com/envoyproxy/envoy/blob/de222c3acf1fbf124ca44f50495fc4b8e9aba525/tools/testdata/protoxform/envoy/v2/oneof.proto.next_major_version_candidate.gold)) in case you wish to make it a oneof in the future.

Reply:
Sounds good. Went ahead and used the oneof_promotion annoatation